### PR TITLE
CAT-225 add leader to record in plaintext output

### PR DESCRIPTION
**Pull request by scoen**
Marcgrep's plain text output format didn't include the Leader for the record. This update fixes this.

Additional changes are:

- nla-deploy script. The app can now be deployed fully from github rather than svn and github 
- configs for test and prod (previously copied in from nla svn)
- tests for the plaintext writer

Original request submitted in DSUP-426

### CODE REVIEW

**Comment by scoen** [comment]
@jrixon Hey Jeremy, I don't suppose you'd be interested in doing a review? :-)
Alex O aside, i think you might be the only other dev with a bit of clojure

**Comment by scoen** [comment]
Ah you might see the version of lucene is downgraded. This is necessary for the app to talk to VuFind's ancient solr index.

**Comment by scoen** [comment]
@ato Hey Alex, was wondering if you'd mind taking a look over this code change i want to make to marcgrep?

File: src/marcgrep/destinations/plaintext.clj [hunk 1]
```diff
@@ -2,13 +2,19 @@
...
 
 (def ms-between-flushes 10000)
 
 (defn write-pretty-record [^Record record ^BufferedWriter out included-fields]
+  (when (or (not included-fields)
+            (included-fields "000"))
+    (let [^Leader lead (.getLeader record)]
+      (do (.write out "000 " )
+          (.write out (.marshal lead))
+          (.write out "\r\n"))))
```
Comment (ato) [inline src/marcgrep/destinations/plaintext.clj]: I think these `(do ...)` forms inside the `let` forms are unnecessary? Maybe I've forgotten something as I'm pretty rusty with Clojure these days but I'm pretty sure `let` allows multiple statements in its body. Not sure why the existing code has it, maybe it was left in accidentally after some refactoring.

```suggestion
      (.write out "000 " )
      (.write out (.marshal lead))
      (.write out "\r\n")))
```

File: test/marcgrep/test/core.clj [hunk 2]
```diff
@@ -245,3 +247,27 @@
...
              :on test-dataset
              :does-not-match ["12345"]))
+
+(deftest write-pretty-record-full
+  "Test writing pretty records: can write complete record including Leader as 000"
+  (let [writer (StringWriter.)
+        rec (marc-record {"001" "111111"} [{:tag "245" :subfields [[\a "test"]]}])
+        match-str "000 00000nam a2200000 a 4500\r\n001 111111\r\n245     $a test\r\n\r\n"]
+    (do (text/write-pretty-record rec writer nil)
+        (is (= match-str (.toString writer))))))
```
Comment (ato) [inline test/marcgrep/test/core.clj]: Same here and in the other two tests:

```suggestion
    (text/write-pretty-record rec writer nil)
    (is (= match-str (.toString writer)))))
```

File: nla-deploy.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -e
+staging_dir=$1
```
Comment (jrixon) [inline nla-deploy.sh]: `staging_dir="$1"`

File: nla-deploy.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+env=$2
```
Comment (jrixon) [inline nla-deploy.sh]: `env="$2"`

**Comment by ato** [comment]
Other than the trivial `do` nit this all looks good to me. Well done. :-)

File: project.clj [hunk 5]
```diff
@@ -6,8 +6,10 @@
                  [compojure "0.6.4"]
                  [ring/ring-servlet "0.3.10"]
                  [org.tigris/marc4j "2.4"]
-                 [org.apache.lucene/lucene-core "5.5.0"]]
-  :plugins [[lein-ring "0.8.5"]]
+                 [org.apache.lucene/lucene-core "3.5.0"]]
```
Comment (jrixon) [inline project.clj:9]: Went from 5.5.0 to 3.5.0?

**Review by jrixon — APPROVED** [review]

File: nla-deploy.sh (hunk 3 above)
Comment (scoen) [reply nla-deploy.sh]: Good idea. Thanks J!

**Review by ato — APPROVED** [review]

File: project.clj (hunk 5 above)
Comment (scoen) [reply project.clj:9]: Yeah, had to drop version so it could talk to old VuFind indexes.

File: nla-deploy.sh (hunk 4 above)
Comment (scoen) [reply nla-deploy.sh]: aye!

**Comment by scoen** [comment]
Thanks for the review and good suggestions - am incorporating them now. Cheers!

**Pull request merged**